src/main/java/io/strimzi/kafka/quotas/StaticQuotaCallback.java

```diff
@@ -145,49 +121,47 @@ public boolean updateClusterMetadata(Cluster cluster) {
         try {
             closeExecutorService();
+            volumeSourceBuilder.close();
         } finally {
-            Metrics.defaultRegistry().allMetrics().keySet().stream().filter(m -> scope.equals(m.getScope())).forEach(Metrics.defaultRegistry()::removeMetric);
+            Metrics.defaultRegistry().allMetrics().keySet().stream().filter(m -> SCOPE.equals(m.getScope())).forEach(Metrics.defaultRegistry()::removeMetric);
         }
     }
 
+    @SuppressWarnings("deprecation")
     @Override
     public void configure(Map<String, ?> configs) {
         StaticQuotaConfig config = new StaticQuotaConfig(configs, true);
         quotaMap = config.getQuotaMap();
-        storageQuotaSoft = config.getSoftStorageQuota();
-        storageQuotaHard = config.getHardStorageQuota();
+        final Optional<Long> availableBytesLimitConfig = config.getAvailableBytesLimit();
+
+        if (availableBytesLimitConfig.isPresent()) {
+            throttleFactorSupplier = new AvailableBytesThrottleFactorSupplier(availableBytesLimitConfig.get());
+            log.info("Available bytes limit {}", availableBytesLimitConfig.get());
+        } else {
+            storageQuotaSoft = config.getSoftStorageQuota();
+            storageQuotaHard = config.getHardStorageQuota();
+
+            throttleFactorSupplier = new TotalConsumedThrottleFactorSupplier(storageQuotaHard, storageQuotaSoft);
+            log.info("Total consumed Storage quota (soft, hard): ({}, {}).", storageQuotaSoft, storageQuotaHard);
+        }
+
+        throttleFactorSupplier.addUpdateListener(() -> resetQuota.add(ClientQuotaType.PRODUCE));
         excludedPrincipalNameList = config.getExcludedPrincipalNameList();
 
         long storageCheckIntervalMillis = TimeUnit.SECONDS.toMillis(config.getStorageCheckInterval());
 
         if (storageCheckIntervalMillis > 0L) {
-            List<Path> logDirs = config.getLogDirs().stream().map(Paths::get).collect(Collectors.toList());
-            storageChecker.configure(
-                    logDirs,
-                    this::updateUsedStorage);
-            backgroundScheduler.scheduleWithFixedDelay(storageChecker, storageCheckIntervalMillis, storageCheckIntervalMillis, TimeUnit.MILLISECONDS);
-            log.info("Configured quota callback with {}. Storage quota (soft, hard): ({}, {}). Storage check interval: {}ms", quotaMap, storageQuotaSoft, storageQuotaHard, storageCheckIntervalMillis);
+            Runnable volumeSource = volumeSourceBuilder.withConfig(config).withVolumeConsumer(throttleFactorSupplier).build();
+            backgroundScheduler.scheduleWithFixedDelay(volumeSource, 0, storageCheckIntervalMillis, TimeUnit.MILLISECONDS);
+            log.info("Configured quota callback with {}. Storage check interval: {}ms", quotaMap, storageCheckIntervalMillis);
+        } else {
+            log.warn("Static quota callback configured to never check usage: set {} to a positive value to enable", StaticQuotaConfig.STORAGE_CHECK_INTERVAL_PROP);
```

Comment:
🤷 The plug-in has been explicitly added to the class path which suggests an intent to monitor but wont do anything as its missing configuration enabling it. 

I'd be happy with either of `WARN` or `INFO`, but landed on `WARN` as that is what the text of the message really is and as it stands out in the logs more.

Reply:
It sounds like this is really, in effect, a required config parameter (if the callback is being used, which be definition is is because `configure()` is being called). If so, then either it should be an `error`, or you should just throw an exception (probably `ConfigException`) with a suitable message.

Reply:
This change is just to add the logging when it's turned off. The actual check and ability to [disable the plug-in](https://github.com/strimzi/kafka-quotas-plugin/blob/main/src/main/java/io/strimzi/kafka/quotas/StaticQuotaCallback.java#L163-L170) is existing behaviour. 

So we can remove support for disabling the plug-in, by enforcing a positive non 0 [value here](https://github.com/strimzi/kafka-quotas-plugin/blob/fc5babec5f77b2090b7633d4bb0d819c8bea7147/src/main/java/io/strimzi/kafka/quotas/StaticQuotaConfig.java#L63l). You can still effectively disable the plug-in by setting it to `Int.MAX_VALUE` though that would still start thread-pools etc if we care about the distinction.

Reply:
The other point I've just realised, and mentioned elsewhere but worth re-iterating here, the plug-in could be used without monitoring disk usage to just apply static quotas. So keeping support for disabling disabling the usage monitoring seems appropriate.

Though it does also mean we should probably drop the log level to `INFO` rather than `WARN` as its a valid configuration.

Reply:
resolved with 86540161ae6062f009eb32af4455b707df179ec7